### DISCUSSION TITLE
How to upgrade from deb package to build from source without losing torrents and settings

### DISCUSSION THREAD

**Discussion by tearfur** [transmission/transmission#3538] [Q&A] [answered]
Anticipating the 4.0.0 release, I may have to switch from the 2.94 deb package to a build from source.

How can I upgrade without losing my torrents? Re-adding and verifying is a pain.

**Comment by ckerr** [comment, marked as answer]
This is probably something that should be put in the release notes.


On non-macOS, you could make a backup of your ~/.config/transmission (or ~/.config/transmission-daemon) directory before upgrading. That way if for whatever reason you don't like the new version, you'll have a backup of the files circa 2.94 to go back to.

Off the top of my head I can't think of anything breaking between 2.94 -> 4.0.0 but 2.94 was a very long time ago, so... when in doubt, make a backup :smile_cat:

**Reply by tearfur** [reply]
The folder people with deb packages are looking for is probably `/var/lib/transmission-daemon/info`.

P.S. On a completely unrelated side note, I can see that this config folder location is specified by `/etc/default/transmission-daemon` if you are using `init`, but with `systemd` I can't figure it out. Is `/var/lib/xxx/info` just where `systemd` stores the configs for its services in general, and applications can obtain this location by a macro or something? Just curious.

**Reply by chaseholden** [reply]
@ckerr It would be great to have mentioned on this thread where to backup macOS settings for Transmission since this is the only search result when doing a search for "Backup" in Discussions section. Kinda shocked it's not here.

Would I be right in guessing that the only settings that need to be backed up for Transmission on macOS are this directory: 
~/Library/Application Support/Transmission  (?)
